File: T1.SqlSharp/T1.SqlSharp/Expressions/SelectColumn.cs
using System.Text;

namespace T1.SqlSharp.Expressions;

public class SelectColumn : ISelectColumnExpression
{
    public SqlType SqlType { get; } = SqlType.SelectColumn;
    public TextSpan Span { get; set; } = new();
    public required ISqlExpression Field { get; set; }
    public string Alias { get; set; } = string.Empty;

    public string ToSql()
    {
        var sql = new StringBuilder();
        sql.Append($"({Field.ToSql()})");
        if (!string.IsNullOrEmpty(Alias))
        {
            sql.Append($" AS {Alias}");
        }
        return sql.ToString();
    }
}

File: VimSharp/chrome-extension-sharp/src/t1-sqlts/Expressions/SelectColumn.ts
import { SqlType } from './SqlType';
import { TextSpan } from '../StringParser';
import { ISqlExpression, SqlVisitor } from './ISqlExpression';
import { ISelectColumnExpression } from './ISelectColumnExpression';

export class SelectColumn implements ISelectColumnExpression {
    SqlType: SqlType = SqlType.SelectColumn;
    Span: TextSpan = new TextSpan();
    Field!: ISqlExpression;
    Alias: string = '';

    Accept(visitor: SqlVisitor): void {
        visitor.Visit_SelectColumn(this);
    }

    ToSql(): string {
        let sql = this.Field.ToSql();
        if (this.Alias) {
            sql += ` AS ${this.Alias}`;
        }
        return sql;
    }
} 